src/main/java/apoc/couchbase/Couchbase.java

```diff
@@ -71,389 +71,338 @@
  * unwind queryResult as queryRow<br/>
  * return queryRow.firstName, queryRow.secondName, queryRow.lastName, queryRow.notableWorks
  * </code>
- *  
- * @since 15.8.2016
+ *
  * @author inserpio
+ * @since 15.8.2016
  */
 public class Couchbase {
 
-  /**
-   * Retrieves a {@link JsonDocument} by its unique ID.
-   * <p/>
-   * Example:
-   * <code>call apoc.couchbase.get(['localhost'], 'default', 'artist:vincent_van_gogh') yield id, expiry, cas, mutationToken, content</code>
-   * 
-   * @param nodes
-   *          the list of nodes to use when connecting to the cluster reference
-   * @param bucket
-   *          the bucket to open; if null is passed then it's used the "default"
-   *          bucket
-   * @param documentId
-   *          the unique ID of the document
-   * @return the found {@link CouchbaseJsonDocument} or null if not found
-   * 
-   * @see Bucket#get(String)
-   */
-  @Procedure
-  @Description("apoc.couchbase.get(nodes, bucket, documentId) yield id, expiry, cas, mutationToken, content - retrieves a couchbase json document by its unique ID.")
-  public Stream<CouchbaseJsonDocument> get(@Name("nodes") List<String> nodes, @Name("bucket") String bucket,
-      @Name("documentId") String documentId){
-    Stream<CouchbaseJsonDocument> result = null;
+    /**
+     * Retrieves a {@link JsonDocument} by its unique ID.
+     * <p/>
+     * Example:
+     * <code>call apoc.couchbase.get('localhost', 'default', 'artist:vincent_van_gogh') yield id, expiry, cas, mutationToken, content</code>
+     *
+     * @param hostOrKey  a URI to use when connecting to the cluster reference or a configuration key
+     * @param bucket     the bucket to open; if null is passed then it's used the "default"
+     *                   bucket
+     * @param documentId the unique ID of the document
+     * @return the found {@link CouchbaseJsonDocument} or null if not found
+     * @see Bucket#get(String)
+     */
+    @Procedure
+    @Description("apoc.couchbase.get(hostOrKey, bucket, documentId) yield id, expiry, cas, mutationToken, content - retrieves a couchbase json document by its unique ID.")
+    public Stream<CouchbaseJsonDocument> get(@Name("hostOrKey") String hostOrKey, @Name("bucket") String bucket,
+                                             @Name("documentId") String documentId) {
+        Stream<CouchbaseJsonDocument> result = null;
 
-    try (CouchbaseConnection couchbaseConnection = getCouchbaseConnection(nodes, bucket)) {
-      JsonDocument jsonDocument = couchbaseConnection.get(documentId);
-      if (jsonDocument != null) {
-        result = Stream.of(new CouchbaseJsonDocument(jsonDocument));
-      }
+        try (CouchbaseConnection couchbaseConnection = getCouchbaseConnection(hostOrKey, bucket)) {
+            JsonDocument jsonDocument = couchbaseConnection.get(documentId);
```

Comment:
I wonder if couchbase has an API that returns a stream/optional or reactive type that we could just `map`.

Reply:
Honestly, I have to take a look because I don't remember.